lib/versioner/simple_test.go

```diff
@@ -93,3 +95,63 @@ func TestSimpleVersioningVersionCount(t *testing.T) {
 		time.Sleep(time.Second)
 	}
 }
+
+func TestPathTildes(t *testing.T) {
+	// Test that folder and version paths with leading tildes are expanded
+	// to the user's home directory. (issue #9241)
+	home := t.TempDir()
+	t.Setenv("HOME", home)
+	t.Setenv("USERPROFILE", home)
+	os.Mkdir(filepath.Join(home, "folder"), 0755)
```

Comment:
This fails on the Windows runner, probably because our legacy HOMEDRIVE/HOMEPATH check takes precedence over USERPROFILE.

Reply:
Ah gotcha, fixed.

Reply:
Hmm, not fixed.
```
\\?\C:\C:\Users\RUNNER~1\AppData\Local\Temp\TestPathTildes1447117282\001\folder\test: The filename, directory name, or volume label syntax is incorrect.
```

Not sure where the extra `C:\` is coming from. Will take another look later today.